api/v1alpha4/machinedeployment_types.go

```diff
@@ -70,6 +70,7 @@ type MachineDeploymentSpec struct {
 	// The deployment strategy to use to replace existing machines with
 	// new ones.
 	// +optional
+	// +kubebuilder:default={type: "RollingUpdate", rollingUpdate: {maxSurge: 1, maxUnavailable: 0}}
```

Comment:
Have you regenerated the manifests to verify that this works?

Reply:
Working on it, my laptop goes on fire when I run it ;/

Reply:
The above seems _very_ error prone. If we add new fields to the struct, or we remove, rename anything, do we need to keep updating this in two places?

Reply:
I think you would have to set it in two places yes, though the CRD generation would blow up (I assume) if it's broken, so we should be able to catch this in CI.

Reply:
Which two places do you mean, webhooks and here? For trivial defaulting such as this, I don't see the need for using the webhook and I'd rather rely on out of the box structural schema.

Reply:
Like @CecileRobertMichon said below:

> +1 on using structural schema for trivial defaults (like a single value), but as soon as there is more complex logic involved, webhooks have the advantage that they can be more easily tested.

The structural schema is ok for single values (strings, enums, or booleans). Let's not use structs defaulting in openapi defaulting, it can become very error prone and hard to debug.

Reply:
mm I'd consider any type with CRD/kubebuilder first class support for defaulting the same trivial use case, being a string or a struct. As a consumer there's no logic involved, you are just assigning your defaults in both cases.

Curious do you have any specific example that would make more obvious `it can become very error prone` that can be related to defaulting MachineDeploymentStrategy as above?

I'll drop it from the PR so we can keep this discussion separately if we want to. Thanks for feedback!

Reply:
@enxebre can you please open an issue to track adding some defaulting + revisiting making Strategy required?

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/4525